From 594b5d56734cd764f3c26d8e849b8ba64b6d1295 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Wed, 3 Dec 2014 18:02:19 +0800
Subject: [PATCH] ARM: atmel: sama5d4: move usb device init code to common
 place

Move usb device initialize code to common place, then it can be
shared by other board which use sama5d4 SoC.

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 arch/arm/cpu/armv7/at91/sama5d4_devices.c | 15 +++++++++++++++
 board/atmel/sama5d4ek/sama5d4ek.c         | 14 +-------------
 2 files changed, 16 insertions(+), 13 deletions(-)

diff --git a/arch/arm/cpu/armv7/at91/sama5d4_devices.c b/arch/arm/cpu/armv7/at91/sama5d4_devices.c
index 4873b50..7469825 100644
--- a/arch/arm/cpu/armv7/at91/sama5d4_devices.c
+++ b/arch/arm/cpu/armv7/at91/sama5d4_devices.c
@@ -7,6 +7,9 @@
 
 #include <common.h>
 #include <asm/io.h>
+#include <asm/arch/at91_common.h>
+#include <asm/arch/at91_pmc.h>
+#include <asm/arch/clk.h>
 #include <asm/arch/sama5d4.h>
 
 char *get_cpu_name()
@@ -29,3 +32,15 @@ char *get_cpu_name()
 	else
 		return "Unknown CPU type";
 }
+
+#ifdef CONFIG_USB_GADGET_ATMEL_USBA
+void at91_udp_hw_init(void)
+{
+	struct at91_pmc *pmc = (struct at91_pmc *)ATMEL_BASE_PMC;
+
+	/* Enable UPLL clock */
+	writel(AT91_PMC_UPLLEN | AT91_PMC_BIASEN, &pmc->uckr);
+	/* Enable UDPHS clock */
+	at91_periph_clk_enable(ATMEL_ID_UDPHS);
+}
+#endif
diff --git a/board/atmel/sama5d4ek/sama5d4ek.c b/board/atmel/sama5d4ek/sama5d4ek.c
index de59011..c1f2703 100644
--- a/board/atmel/sama5d4ek/sama5d4ek.c
+++ b/board/atmel/sama5d4ek/sama5d4ek.c
@@ -107,18 +107,6 @@ static void sama5d4ek_usb_hw_init(void)
 }
 #endif
 
-#ifdef CONFIG_USB_GADGET_ATMEL_USBA
-static void sama5d4ek_udp_hw_init(void)
-{
-	struct at91_pmc *pmc = (struct at91_pmc *)ATMEL_BASE_PMC;
-
-	/* Enable UPLL clock */
-	writel(AT91_PMC_UPLLEN | AT91_PMC_BIASEN, &pmc->uckr);
-	/* Enable UDPHS clock */
-	at91_periph_clk_enable(ATMEL_ID_UDPHS);
-}
-#endif
-
 #ifdef CONFIG_LCD
 vidinfo_t panel_info = {
 	.vl_col = 800,
@@ -319,7 +307,7 @@ int board_init(void)
 	sama5d4ek_mci1_hw_init();
 #endif
 #ifdef CONFIG_USB_GADGET_ATMEL_USBA
-	sama5d4ek_udp_hw_init();
+	at91_udp_hw_init();
 #endif
 #ifdef CONFIG_MACB
 	sama5d4ek_macb0_hw_init();
-- 
1.9.1

